src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { FC, useEffect, useState } from 'react';
+
+type ProgressBarProps = {
+  page: number;
+  totalPages: number;
+};
+
+export const ProgressBar: FC<ProgressBarProps> = ({ page, totalPages }) => {
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    setProgress((page / totalPages) * 100);
+  }, [page, totalPages]);
```

Comment:
This can be simplified as we don't have to rely on any "effects" - aka network calls or similar.

```suggestion
  const progress = (page / totalPages) * 100;
```